## Final Project Submission

Please fill out:
* Student name: Dalton Rabb
* Student pace: full time
* Scheduled project review date/time: 3/26
* Instructor name: James Irving
* Blog post URL:


## Microsoft Studios Pitch Analysis

In [78]:
# Your code here - remember to use markdown cells for comments as well!

In [79]:
import pandas as pd
import numpy as np

In [80]:
basics = pd.read_csv('zippeddata/imdb.title.basics.csv.gz')
ratings = pd.read_csv('zippeddata/imdb.title.ratings.csv.gz')
gross = pd.read_csv('zippeddata/bom.movie_gross.csv.gz')
popularity = pd.read_csv('zippeddata/tmdb.movies.csv.gz')
budgets = pd.read_csv('zippeddata/tn.movie_budgets.csv.gz')

In [81]:
basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


In [82]:
ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   tconst         73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


In [83]:
gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [84]:
basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

In [85]:
basics['genres'].value_counts()[:10]

Documentary              32185
Drama                    21486
Comedy                    9177
Horror                    4372
Comedy,Drama              3519
Thriller                  3046
Action                    2219
Biography,Documentary     2115
Drama,Romance             2079
Comedy,Drama,Romance      1558
Name: genres, dtype: int64

In [86]:
popularity.describe()

Unnamed: 0             id    popularity  vote_average    vote_count
count  26517.00000   26517.000000  26517.000000  26517.000000  26517.000000
mean   13258.00000  295050.153260      3.130912      5.991281    194.224837
std     7654.94288  153661.615648      4.355229      1.852946    960.961095
min        0.00000      27.000000      0.600000      0.000000      1.000000
25%     6629.00000  157851.000000      0.600000      5.000000      2.000000
50%    13258.00000  309581.000000      1.374000      6.000000      5.000000
75%    19887.00000  419542.000000      3.694000      7.000000     28.000000
max    26516.00000  608444.000000     80.773000     10.000000  22186.000000

In [87]:
budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [88]:
budgets['worldwide_gross'] = budgets['worldwide_gross'].str.replace(',', '')
budgets['worldwide_gross'] = budgets['worldwide_gross'].str.replace('$', '')
budgets['production_budget'] = budgets['production_budget'].str.replace(',', '')
budgets['production_budget'] = budgets['production_budget'].str.replace('$', '')


In [89]:
print(budgets['worldwide_gross'])

0       2776345279
1       1045663875
2        149762350
3       1403013963
4       1316721747
           ...    
5777             0
5778        240495
5779          1338
5780             0
5781        181041
Name: worldwide_gross, Length: 5782, dtype: object


In [90]:
wwg_list = budgets['worldwide_gross'].tolist()
wwg_list = list(map(int, wwg_list))

In [91]:
wwg_list

[2776345279,
 1045663875,
 149762350,
 1403013963,
 1316721747,
 2053311220,
 2048134200,
 963420425,
 655945209,
 879620923,
 1084439099,
 393151347,
 260002115,
 282778100,
 586477240,
 894860230,
 1140069413,
 867500281,
 1017003568,
 935213767,
 960366855,
 945577621,
 1234846267,
 374085065,
 788241137,
 591692078,
 1517935897,
 1066215812,
 667999518,
 417341288,
 757890267,
 313477717,
 602893340,
 1648854864,
 654213485,
 836519699,
 1104039076,
 459260946,
 322459006,
 550517357,
 367262558,
 1348258224,
 2208208395,
 1242520711,
 1049102856,
 1021215193,
 1068879522,
 1215392272,
 862316233,
 795110670,
 1025491110,
 1110526981,
 743588329,
 490359051,
 747862775,
 708996336,
 560155383,
 400062763,
 757677748,
 652220086,
 365491792,
 219535492,
 336359676,
 1123790543,
 409953905,
 197687603,
 1518722794,
 467381584,
 531514650,
 351040419,
 315709697,
 411002906,
 333771037,
 585171547,
 1001996207,
 786635413,
 554606532,
 561072059,
 335802233,
 259357408,
 619234314,
 8

In [92]:
pb_list = budgets['production_budget'].tolist()
pb_list = list(map(int, pb_list))

In [93]:
pb_list

[425000000,
 410600000,
 350000000,
 330600000,
 317000000,
 306000000,
 300000000,
 300000000,
 300000000,
 300000000,
 275000000,
 275000000,
 275000000,
 275000000,
 260000000,
 258000000,
 250000000,
 250000000,
 250000000,
 250000000,
 250000000,
 250000000,
 250000000,
 232000000,
 230000000,
 230000000,
 225000000,
 225000000,
 225000000,
 225000000,
 220000000,
 220000000,
 217000000,
 215000000,
 215000000,
 210000000,
 210000000,
 210000000,
 210000000,
 207000000,
 205000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 200000000,
 195000000,
 195000000,
 195000000,
 190000000,
 190000000,
 190000000,
 190000000,
 190000000,
 190000000,
 187500000,
 186000000,
 185000000,
 185000000,
 185000000,
 185000000,
 185000000,
 185000000,
 182000000,
 180000000,
 180000000,
 180

In [94]:
wwg_array = np.array(wwg_list)
pb_array = np.array(pb_list)

In [95]:
roi_array = wwg_array / pb_array

In [96]:
roi_array = roi_array * 100

In [111]:
roi_list = roi_array.round(2).tolist()
roi_list

[653.26,
 254.67,
 42.79,
 424.38,
 415.37,
 671.02,
 682.71,
 321.14,
 218.65,
 293.21,
 394.34,
 142.96,
 94.55,
 102.83,
 225.57,
 346.85,
 456.03,
 347.0,
 406.8,
 374.09,
 384.15,
 378.23,
 493.94,
 161.24,
 342.71,
 257.26,
 674.64,
 473.87,
 296.89,
 185.49,
 344.5,
 142.49,
 277.83,
 766.91,
 304.29,
 398.34,
 525.73,
 218.7,
 153.55,
 265.95,
 179.15,
 674.13,
 1104.1,
 621.26,
 524.55,
 510.61,
 534.44,
 607.7,
 431.16,
 397.56,
 512.75,
 555.26,
 371.79,
 245.18,
 373.93,
 354.5,
 280.08,
 200.03,
 378.84,
 326.11,
 182.75,
 109.77,
 168.18,
 576.3,
 210.23,
 101.38,
 799.33,
 245.99,
 279.74,
 184.76,
 166.16,
 216.32,
 178.01,
 314.61,
 541.62,
 425.21,
 299.79,
 303.28,
 181.51,
 140.19,
 340.24,
 470.54,
 400.3,
 421.41,
 445.78,
 295.84,
 142.55,
 193.83,
 100.03,
 119.5,
 101.67,
 442.39,
 304.8,
 297.49,
 208.17,
 641.75,
 550.2,
 488.13,
 502.95,
 426.32,
 417.98,
 456.0,
 299.59,
 218.11,
 219.17,
 172.84,
 126.42,
 231.86,
 99.5,
 151.0,
 79.79,
 86.7,
 768.1,
 453

In [112]:
budgets['roi'] = roi_list 

In [113]:
budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross     roi  
0         425000000   $760,507,625      2776345279  653.26  
1         410600000   $241,063,875      1045663875  254.67  
2         350000000    $42,762,350       149762350   42.79  
3         330600000   $459,005,868      1403013963  424.38  
4         317000000   $620,181,382      1316721747  415.37

In [101]:
genre_ratings_df = pd.merge(basics,
                            ratings,
                            on = ['tconst'],
                           how = 'left')

In [102]:
genre_ratings_df.nlargest(100, 'numvotes').sort_values(by=['numvotes'], ascending = False)

tconst                    primary_title  \
7066   tt1375666                        Inception   
6900   tt1345836            The Dark Knight Rises   
311    tt0816692                     Interstellar   
20342  tt1853728                 Django Unchained   
356    tt0848228                     The Avengers   
...          ...                              ...   
170    tt0470752                       Ex Machina   
12366  tt1659337  The Perks of Being a Wallflower   
18045  tt1800241                  American Hustle   
48318  tt2872718                     Nightcrawler   
6395   tt1291584                          Warrior   

                        original_title  start_year  runtime_minutes  \
7066                         Inception        2010            148.0   
6900             The Dark Knight Rises        2012            164.0   
311                       Interstellar        2014            169.0   
20342                 Django Unchained        2012            165.0   
356                       The Avengers        2012            143.0   
...                                ...         ...              ...   
170                         Ex Machina        2014            108.0   
12366  The Perks of Being a Wallflower        2012            103.0   
18045                  American Hustle        2013            138.0   
48318                     Nightcrawler        2014            117.0   
6395                           Warrior        2011            140.0   

                        genres  averagerating   numvotes  
7066   Action,Adventure,Sci-Fi            8.8  1841066.0  
6900           Action,Thriller            8.4  1387769.0  
311     Adventure,Drama,Sci-Fi            8.6  1299334.0  
20342            Drama,Western            8.4  1211405.0  
356    Action,Adventure,Sci-Fi            8.1  1183655.0  
...                        ...            ...        ...  
170       Drama,Mystery,Sci-Fi            7.7   424542.0  
12366                    Drama            8.0   422671.0  
18045              Crime,Drama            7.2   418221.0  
48318     Crime,Drama,Thriller            7.9   404476.0  
6395               Drama,Sport            8.2   403755.0  

[100 rows x 8 columns]

From the merged data set we can see popularity, based off number of votes and average rating along with the genres of the title. 

In [103]:
genre_ratings_df.nlargest(100, 'numvotes').mean()

start_year           2013.250
runtime_minutes       127.990
averagerating           7.671
numvotes           606520.830
dtype: float64

In [104]:
genre_ratings_df.nlargest(100, 'numvotes').std()

start_year              2.212796
runtime_minutes        20.002270
averagerating           0.517823
numvotes           224778.932318
dtype: float64

In [105]:
genre_ratings_df['averagerating'].mean()

6.332728552859617

Here we can see the average run time among the top 100 most popular movies is 128 minutes, with a standard deviation of 20 minutes. The average rating is 7.67, which is 1.34 points higher than the average.

In [106]:
import matplotlib.ticker as mtick

budgets['']


KeyError: ''